### PR TITLE
Improv/inject headers in logs

### LINKED ISSUE #6: No logs on most error cases
**Problem statement**
There are multiple cases where errors are not being logged. Also, despite the fact the log messages are descriptive they don't provide enough context to help identify the root cause.

**Proposed solution**
1. Detailed error messages should be logged every time a HTTP error is returned. 
2. Allow optional injection of http request headers

_Plugin configuration suggestion_
```yaml
keysurl: "https://jwks.keys"
issuer: "https://issuer.test"
audience: "audience1,audience2"
roles:
  - Test.Role.1
  - Test.Role.2
matchallroles: true
loglevel: "DEBUG"
logheaders: 
  - X-Request-Id
  - Host
```
Having this would allow easier troubleshooting in error scenarios.

### PR DESCRIPTION
**Pull request by t1agob**
closes #6.

- Adds support for _logHeaders_ configuration
- Adds support for adding HTTP headers to HTTP error logs
- Adds tests to cover added code
- Adds documentation explaining how to use new parameters

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ This is a Traefik plugin which validates JWT tokens generated by Azure Active Di
 |roles | List of roles to be validated by the plugin. |*string[]*| no  |
 |matchallroles | Flag to let plugin know if all roles need to be matched to return success. |*boolean*| no  |
 |loglevel | Log level for plugin execution. Defaults to *'WARN'* level. <br /><br /> **Possible Values:** INFO, WARN, DEBUG.  |*string*| no  |
+|logheaders | If enabled, HTTP headers are added to logs. Defaults to 'no'. |*boolean*| no  |
+|filterheaders | If specified, only this HTTP headers will be added to the logs. Defaults to adding all headers to the logs. |*string[]*| no  |
```
Comment (Herbievorious) [inline README.md:22]: `only this HTTP headers` should be `only these HTTP headers`

Comment (michdim) [inline README.md]: these

File: validatetoken.go [hunk 2]
```diff
@@ -335,3 +338,27 @@ func (claims *Claims) isValidForRole(configRole string) bool {
...
+			}
+		}
+	}
+
+	logPayload["StatusCode"] = strconv.Itoa(statusCode)
+	logPayload["Url"] = request.URL.String()
+	logPayload["Method"] = request.Method
+	logPayload["Error"] = message
+
+	jsonStr, _ := json.Marshal(logPayload)
```
Comment (michdim) [inline validatetoken.go:362]: It's good practice to handle the error here:
jsonStr, err := json.Marshal(logPayload)
    if err != nil {
        logger.Printf("Error marshaling log payload to JSON: %v\n", err)
        return
    }

Comment (t1agob) [reply validatetoken.go]: fixed now!

**Pull request merged**